src/redux/books/books.js

```diff
@@ -1,29 +1,86 @@
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const GET_CURRENT_API = 'bookStore/books/GET_CURRENT_API';
+const GET_CURRENT_API_SUCCESS = 'bookStore/books/GET_CURRENT_API_SUCCESS';
+const GET_CURRENT_API_FAILURE = 'bookStore/books/GET_CURRENT_API_FAILURE';
+const API = `
+  https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/0ePu6tv5E4TBgiHo9yos/books`;
 
-const initialState = [];
+const initialState = {
+  status: 'Idle',
+  books: {},
+};
+
+export const getApi = async (dispatch) => {
+  dispatch({ type: GET_CURRENT_API });
+  const books = await fetch(API).then((res) => res.json());
+  return dispatch({ type: GET_CURRENT_API_SUCCESS, books });
+};
```

Comment:
- [ ] As requested of you by the previous reviewer, kindly note that it is a requirement that you make use of [createasyncthunk](https://redux-toolkit.js.org/api/createAsyncThunk) when making your API calls according to the project requirements. I figured out that it was hard for you to implement this required change, I will give you an example trying to explain the moving parts as best and as briefly as I can.

```
import { createAsyncThunk } from '@reduxjs/toolkit';

export const getApi = createAsyncThunk(GET_CURRENT_API, async () => {
  const response = await fetch(API);
  const data = await response.json();

  const newArray = Object.keys(data).map((id) => {
    const obj = data[id][0];
    obj.id = id;
    return obj;
  });

  console.log(newArray);

  return newArray;
});
```

**Explanation**

1. You are first importing the createAsyncThunk into your file to use it. Make sure that you install it first.
2. You are then going to create the getBooks method using the createAsyncThunk. Here the GET_BOOK is simply the action type, nothing fancy
3. Then, because the data returned is of object data structure, I want to convert it into an array and return that array. This step is completely not a must and you could skip it and just return data.

I hope this very brief explanation helps you. Removing and adding a book with follow the same pattern as well.